galaxy_ng/app/management/commands/analytics/galaxy_collector.py

```diff
@@ -189,42 +183,80 @@ def container_tag_table(since, full_path, until, **kwargs):
     "galaxy_legacynamespace", "1.0", format="csv", description="Data on galaxy_legacynamespace"
 )
 def galaxy_legacynamespace_table(since, full_path, until, **kwargs):
-    source_query = """COPY (SELECT
+    source_query = """
+        SELECT
             id, created, modified, name, company, avatar_url, description, namespace_id
-            FROM galaxy_legacynamespace
-        ) TO STDOUT WITH CSV HEADER"""
+        FROM galaxy_legacynamespace
+    """
     return _simple_csv(full_path, "galaxy_legacynamespace", source_query)
 
 
 @register("galaxy_legacyrole", "1.0", format="csv", description="Data on galaxy_legacyrole")
 def galaxy_legacyrole_table(since, full_path, until, **kwargs):
-    source_query = """COPY (SELECT
+    source_query = """
+        SELECT
             id, created, modified, name, full_metadata, namespace_id
-            FROM galaxy_legacyrole
-        ) TO STDOUT WITH CSV HEADER"""
+        FROM galaxy_legacyrole
+    """
     return _simple_csv(full_path, "galaxy_legacyrole", source_query)
 
 
 @register(
     "galaxy_aiindexdenylist", "1.0", format="csv", description="Data on galaxy_aiindexdenylist"
 )
 def galaxy_aiindexdenylist_table(since, full_path, until, **kwargs):
-    source_query = """COPY (SELECT * FROM galaxy_aiindexdenylist
-        ) TO STDOUT WITH CSV HEADER"""
+    source_query = """
+        SELECT * FROM galaxy_aiindexdenylist
+    """
     return _simple_csv(full_path, "galaxy_aiindexdenylist", source_query)
 
 
 def _get_csv_splitter(file_path, max_data_size=209715200):
     return CsvFileSplitter(filespec=file_path, max_file_size=max_data_size)
 
 
-def _simple_csv(full_path, file_name, query, max_data_size=209715200):
+def _simple_csv(
+    full_path: PosixPath,
+    file_name: str,
+    query: str,
+    max_data_size: int = 209715200
+) -> List[str]:
+    """
+    Execute a psql query and write to a splitted temp file.
+
+    This function previously used cursor.copy_expert() previously
+    to emit a csv structure directly to stdout but that function
+    no longer exists in django 4.x + psycopg which will come from
+    upgrading to pulp >= 3.25.
+
+    Args:
+        full_path: A PosixPath temp directory were the files are written.
+        file_name: A string file basename where the output is written.
+        query: The psql query to run and save the output of.
+        max_data_size: An integer that controls the csv splitter.
+
+    Returns:
+        A list of strings that are filepaths where the data
+        was written to. It is usually just one filepath.
+    """
     file_path = _get_file_path(full_path, file_name)
     tfile = _get_csv_splitter(file_path, max_data_size)
 
+    # use a normal .execute to get the data as rows ...
     with Collector.db_connection().cursor() as cursor:
-        cursor.copy_expert(query, tfile)
```

Comment:
see this article on how to write copy_expert in psycopg3 https://www.psycopg.org/articles/2020/11/15/psycopg3-copy/

Then the rest of this PR shouldn't be needed. (we can't actually do it that way to avoid OOM, also COPY command is much more effective than basic select)